File: Backend/api/predictionInputRoutes.js
const express = require('express');
const PredictionInputController = require('../controllers/predictionInputController');
const auth = require('../middlewares/auth');

const router = express.Router();
router.use(auth);
router.get('/', PredictionInputController.getAll);
router.get('/stats', PredictionInputController.getStats);

module.exports = router;

File: backend_python/api/prediction_input_routes.py
from flask import Blueprint

from controllers.prediction_input_controller import PredictionInputController
from middlewares.auth import auth_required

bp = Blueprint("prediction_inputs", __name__)

bp.add_url_rule("/", endpoint="get_all", view_func=auth_required(PredictionInputController.get_all), methods=["GET"])
bp.add_url_rule("/stats", endpoint="stats", view_func=auth_required(PredictionInputController.get_stats), methods=["GET"])
